REDCapUITweaker.php

```diff
@@ -202,6 +202,30 @@ function redcap_every_page_before_render( $project_id = null )
 				$GLOBALS['lang']['dataqueries_309'] = $dqCustomBodyDRW;
 			}
 
+			// If a submission which includes @SQLCHECKBOX fields, ensure that the selected options
+			// are allowed to be submitted.
+			$listValuesSQLChkbx = [];
+			if ( $_GET['page'] != '' && ! empty( $_POST ) &&
+			     $this->getSystemSetting( 'sql-checkbox' ) )
+			{
+				foreach ( $_POST as $postField => $postVal )
+				{
+					if ( substr( $postField, -22 ) == ':uitweaker-sqlcheckbox' )
```

Comment:
Is this always at the end of $postField?  may not another item submitted be added after it, in which case you would want to look anywhere in the string?

Reply:
For a given SQLCHECKBOX field `field_name`, the JavaScript will add a hidden field `field_name:uitweaker-sqlcheckbox` to the field when the checkboxes are loaded, so it is flagged as a SQLCHECKBOX field. Thus it is always at the end of the field name here.